### PR TITLE
Remove unused B2G-only stream_type field from cubeb_stream_params.

### PR DESCRIPTION
**Pull request by kinetiknz**
See https://github.com/kinetiknz/cubeb/issues/346

### CODE REVIEW

File: src/cubeb_opensl.c [hunk 1]
```diff
@@ -1241,29 +1214,6 @@ opensl_configure_playback(cubeb_stream * stm, cubeb_stream_params * params) {
...
-    SLAndroidConfigurationItf playerConfig;
-    res = (*stm->playerObj)->GetInterface(stm->playerObj,
-                                          stm->context->SL_IID_ANDROIDCONFIGURATION,
-                                          &playerConfig);
-    if (res != SL_RESULT_SUCCESS) {
-      LOG("Failed to get android configuration interface. Error code: %lu", res);
-      return CUBEB_ERROR;
-    }
-
-    res = (*playerConfig)->SetConfiguration(playerConfig,
```
Comment (kinetiknz) [inline src/cubeb_opensl.c]: SL_ANDROID_STREAM_MEDIA (== CUBEB_STREAM_TYPE_MUSIC) is the default according to https://developer.android.com/ndk/guides/audio/opensl/android-extensions.html

**Comment by kinetiknz** [comment]
@achronop Do you mind reviewing this as well? Mainly for the Android bits and just so you're aware we're doing this.

**Comment by achronop** [comment]
Why did you decide to remove that? It may be better to try to use it in gecko side rather than remove it. For example we could use MEDIA for WebAudio or playback and VOICE_CALL for WebRTC. I see that chrome does something similar:
https://cs.chromium.org/chromium/src/media/audio/android/audio_manager_android.cc?l=234

**Comment by achronop** [comment]
I reviewed it, it's good to go, I do not have any comment just the question above.

**Comment by kinetiknz** [comment]
> Why did you decide to remove that?

It was added in a rush and never properly designed, leaving us with the ugly hack of a platform-specific field in the public API and exposing the Android/B2G stream type constants directly without consideration of how they are handled on other platforms.

The code is dead and untested since B2G development stopped.  Keeping dead code around has real maintenance costs and zero benefits.

> It may be better to try to use it in gecko side rather than remove it. For example we could use MEDIA for WebAudio or playback and VOICE_CALL for WebRTC.

If you think it's worth supporting a better designed (i.e. portable) and properly maintained implementation of this feature, file a bug in BMO and make a case for it there.

**Comment by padenot** [comment]
I agree with @kinetiknz here, this is not ideal and was done too quickly. In practice, the following needs to happen:
- Have specs recognize the fact that authors need to be able to express intent when doing something that will end up creating an audio stream (i.e., is this AudioContext is going to be used for speech, music, something else ?)
- Make a proper API in cubeb, that will abstract this over different platforms (on Windows, you can use the "communication device", and on Android "VOICE_CALL" as noted. iirc there is something similar for pulse, and we could use a `kAudioUnitSubType_VoiceProcessingIO` on OSX)

**Pull request merged**